packages/react-topology/src/components/edges/terminals/ConnectorArrow.tsx

```diff
@@ -25,10 +25,6 @@ const ConnectorArrow: React.FunctionComponent<ConnectorArrowProps> = ({
   size = 14,
   dragRef
 }) => {
-  if (!startPoint || !endPoint) {
-    return null;
-  }
-
```

Comment:
Can we be sure `startPoint` and `endPoint` will never be null? I think we need safety checks below if we are passing thru with potentially null values.

Reply:
This is a good callout. Simply moving this condition just above the other `return` should alleviate the issue of conditional hooks and maintain the existing logic.

Reply:
According to the type definition it should always be defined as these props are marked as required there. Naturally this does not prevent non-TypeScript users from forgetting to pass in these props so you might want to take that into account.

It's not possible to move the conditional around as `startPoint` and `endPoint` are both used immediately underneath, even before the `useMemo()`.

So this would mean that either we have the following options AFAIK:

1. Leave this and let TypeScript catch the issues, non-TS users will get a runtime error.
2. Move the logic into a seperate component and render this after the check for startPoint` and `endPoint`.

Reply:
We just need to check them inside the useMemo call, and then return if null before continuing:

```
const ConnectorArrow: React.FunctionComponent<ConnectorArrowProps> = ({
  startPoint,
  endPoint,
  className = '',
  size = 14,
  dragRef
}) => {
  const polygonPoints = React.useMemo(
    () =>
      pointsStringFromPoints([
        [0, size / 2],
        [0, -size / 2],
        [size, 0]
      ]),
    [size]
  );
  const boundingPoints = React.useMemo(() => {
    if (startPoint || !endPoint) {
      return null;
    }
    return pointsStringFromPoints(getConnectorBoundingBox(startPoint, endPoint, size));
  }, [endPoint, size, startPoint]);

  if (!startPoint || !endPoint) {
    return null;
  }

  const connectorStartPoint = getConnectorStartPoint(startPoint, endPoint, size);
  const angleDeg = getConnectorRotationAngle(startPoint, endPoint);

  const classNames = css(styles.topologyConnectorArrow, className, dragRef && 'pf-m-draggable');

  return (
    <g
      transform={`translate(${connectorStartPoint[0]}, ${connectorStartPoint[1]}) rotate(${angleDeg})`}
      ref={dragRef}
      className={classNames}
    >
      <polygon points={polygonPoints} />
      <polygon points={boundingPoints} fillOpacity={0} strokeWidth={0} />
    </g>
  );
};
```

Reply:
I like this approach, I've dded a commit with this change @jeff-phillips-18.